components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/mutualSSL_handler.bal

```diff
@@ -58,98 +58,155 @@ public type MutualSSLHandler object {
         if (mutualSSLVerifyClient is string && stringutils:equalsIgnoreCase(MANDATORY, mutualSSLVerifyClient) 
                 && req.mutualSslHandshake[STATUS] != PASSED) {
             if (req.mutualSslHandshake[STATUS] == FAILED) {
-                printDebug(KEY_AUTHN_FILTER, "MutualSSL handshake status: FAILED");
+                printError(KEY_AUTHN_FILTER, "MutualSSL handshake status: FAILED");
             }
             // provided more generic error code to avoid security issues.
             setErrorMessageToInvocationContext(API_AUTH_INVALID_CREDENTIALS); 
             return prepareAuthenticationError("Failed to authenticate with MutualSSL handler");            
         }
         if (req.mutualSslHandshake[STATUS] == PASSED) {
+            boolean | http:AuthenticationError mutualSSLStatus = false;
             runtime:InvocationContext invocationContext = runtime:getInvocationContext();
             if (mutualSSLVerifyClient is string && stringutils:equalsIgnoreCase(MANDATORY, mutualSSLVerifyClient)) {
                 string apiVersion = invocationContext.attributes[API_VERSION_PROPERTY].toString();
                 string apiName = invocationContext.attributes[API_NAME].toString();
                 if (self.headerName != "" &&  req.hasHeader(self.headerName)) {
+                    printDebug(KEY_AUTHN_FILTER, "Mutual ssl expected header " + self.headerName + " present in the request");
+                    //If validation is disabled for client certificate present in the context , we should always
+                    //validate the certificate present in the header.
+                    //This scenario represents where mtls is required between client and mgw, but mtls is not enabled
+                    //between LB and mgw. So microgateway only validates the certificate present in the header, which
+                    //is the client certificate not the certificate in the context which is always going to be
+                    //LB certificate but will not be available due to no MTLS between mgw and LB.
```

Comment:
isClientCertificateValidationEnabled config is only valid when certificate is present in the header. If header is not present we always validate certificate in the context. 
Whats the confusion in the comment

Reply:
if isClientCertificateValidationEnabled is applicable for the header, can we mention about the header in the comment. Otherwise, it looks like this is intended for certificatevalidation disabling for both header and context as well